### ISSUE TITLE
Reproduction gap in semantic segmentation

### ISSUE THREAD

**Issue by wsgbxx** [facebookresearch/pixio#8]
Hi , thanks for releasing the code and checkpoints!

I’m reproducing the downstream tasks in this repo. Monocular depth estimation (frozen encoder + trainable head) matches the paper closely on my side, but I’m seeing a large gap for semantic segmentation (semseg) and occasional NaNs in the training loss, even though I tried to follow the semseg scripts and hyperparameters in the repository as closely as possible.

Expected results (from paper)

<img width="632" height="362" alt="Image" src="https://github.com/user-attachments/assets/456a4804-e35b-430f-9ca8-d45638497523" />

From Table 4, for frozen encoder + DPT head the reported mIoU values are:

DINOv3-H+: ADE20K 52.3, LoveDA 55.3

Pixio-H: ADE20K 53.6, LoveDA 54.7 

The paper also mentions the semseg protocol as:

1.DPT head uses 4 intermediate features evenly from 4 encoder stages

2.60 epochs training

3.training resolution 256×256 for patch size 16 encoders (224×224 for patch size 14)

4.inference uses sliding window with overlap + ensembling

My reproduction results (semseg)

ADE20K：

DINOv3 + DPT head: ~48 mIoU 

Pixio-H + DPT head: ~48 mIoU

Pixio-H + linear head: ~45 mIoU 

LoveDA：

DINOv3 + DPT head: ～40 mIoU

Pixio-H + DPT head:  ～38 mIoU

Pixio-H + DPT head:  ～39 mIoU

(For LoveDA I also observed strong instability early on; after some mitigation it may stabilize but still far from paper.)

During semseg training I also frequently see NaNs in the loss (e.g., after some iterations/epochs). Once NaNs occur, the run becomes unstable and the final mIoU degrades significantly.

What I have tried / checked：

1.Used the repo’s semseg training/eval scripts and configs as-is .

2.Confirmed the encoder weights load correctly and the encoder is frozen (only training the head).

3.Tried to match the paper protocol: 60 epochs, 256×256 for patch16 encoders, sliding-window inference with overlap/ensembling.

4.Experimented with GN, since BN/SyncBN with small effective batch size can cause instability; this reduced the drift in some cases but the final mIoU is still much lower than reported.

5.Tried to increase cropsize to 512  and batchsize to 2 ( per GPU * 8 GPUs)

If you’d like, I can provide full logs and the exact commands/configs I used. Any pointers on what might be missing for semseg reproduction would be very helpful. Thanks!

**Comment by LiheYoung** [comment]
Hi, thanks for your interest.

It would be nice if you could share the logs.

**Comment by wsgbxx** [comment]
Thank you for reply!
For the simple pixio in lovede with linear heads 
[logs.txt](https://github.com/user-attachments/files/24675127/logs.txt)

<img width="2140" height="961" alt="Image" src="https://github.com/user-attachments/assets/c9409321-0218-45b8-b2ec-8f744e6625ef" />

Thanks again — please let me know if there’s any other specific info you’d like me to include.

**Comment by LiheYoung** [comment]
For LoveDA, did you pre-process the masks as the instructions in README:
```bash
python evaluation/semseg/util/process_loveda.py
```

For ADE20K, did you set the batch size as 16 (2 imgs/gpu x 8 gpus) and use the provided pre-processed ground-truth masks?

**Comment by wsgbxx** [comment]
Thank you very much.
For LoveDA and ADE20K,my data sets did not be pre-processed.
The new results match the paper closely.
